pkg/volume/csi/csi_plugin.go

```diff
@@ -796,11 +786,72 @@ func (p *csiPlugin) skipAttach(driver string) (bool, error) {
 	return false, nil
 }
 
+// supportsVolumeMode looks up CSIDriver object associated with the
+// driver name to determine if the driver supports the given volume
+// mode. An error indicates that it isn't supported and explains why.
+func (p *csiPlugin) supportsVolumeMode(driver string, volumeMode storage.CSIVolumeMode) error {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume) {
+		// Nothing to check, feature disabled.
+		return nil
```

Comment:
Should this actually return success only when volumeMode is Persistent?  If a user wants ephemeral mode, but then the cluster gets downgraded to a version where the feature is disabled, we probably don't want to continue mounting the volume as persistent.

Reply:
You are right. So I'll revert to the version I had before: `supportsVolumeMode` always has to be called, and depending on the current volume mode and the feature flag it'll decide whether kubelet is meant to go ahead.

Reply:
`supportsVolumeMode` rewritten, please have another look.